content/developer/howtos/rdtraining/02_setup.rst

```diff
@@ -460,5 +470,103 @@ Here is a list of commands:
+- If the Atlassian tutorial does not work for you, W3schools offers a
+  `very nice alternative <https://www.w3schools.com/git/>`__.
+- To visualize the effect of git commands on the commits graphs, play with this
+  `interactive tutorial <https://learngitbranching.js.org/>`__.
+- If you want to read more about Github, their
+  `documentation <https://docs.github.com/en/get-started/quickstart/hello-world>`__
+  includes a comprehensive introduction.
+
+Python
+^^^^^^
+
+- Don't forget about `Python official documentation <https://docs.python.org/3/>`__
+  (and make sure to select the correct version at the top of the page!).
+- `The Hitchhiker's Guide <https://docs.python-guide.org/>`__ will teach you the good practices of
+  Python.
+- If you have good experience of similar languages or just need a quick recall,
+  `this guide <https://learnxinyminutes.com/docs/python3/>`__ is made for you.
+
+Javascript
+^^^^^^^^^^
+
+- You should find a tutorial that suits your level in this `"re-introduction" to JavaScript
+  <https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript>`__.
+- If you just need a quick recall, `this one <https://learnxinyminutes.com/docs/javascript/>`__
+  will do the trick.
+
+(Postgre)SQL
+^^^^^^^^^^^^
+
+- If you are not familiar with PostgreSQL, `these exercises <https://www.pgexercises.com/>`__
+  will teach you better than any long documentation.
+- Even though you shouldn't have to interact with it, you might be interested to know
+  that Odoo uses `psycopg2 <https://www.psycopg.org/docs/usage.html>`__ to interact
+  with its SQL backend.
+
+Debugging
+^^^^^^^^^
+
+At first, a simple `print(...)` statement here and there might do the trick. After a while
+however, learning how to use a proper debugger will make the experience much better for you.
+
+In the `Debugging page <https://github.com/odoo/enterprise/wiki/Debugging>`__,
+we present two of the most common debuggers to track down bugs in Python,
+and how you can use them with Odoo.
+
+Mixins
+^^^^^^
+
+If you need to interface with common Odoo features such as the chatter, you can rely on
+`mixins <https://www.odoo.com/documentation/master/developer/reference/mixins.html>`__.
+They are Odoo models exposing useful methods through inheritance.
+
+To learn and play with mixins, visit
+`this repository <https://github.com/tivisse/odoodays-2018/>`__.
+This module for a plant nursery is training material developed for the OXP 2018.
+You don't need to code it on your side. But you can check the presentations in the `/static/pdf`
+directory and play with the module to discover some magic features in Odoo.
+
+Security
+^^^^^^^^
+
+Every day we hear about data leaks and attacks in the newspapers.
+We expect you to be aware and careful about how to avoid security breaches in your future
+developments.
+
+You will define new models during your training, and there are some things
+at which you should pay extra attention:
```

Comment:
Since this section will move to advanced topics, the sentence can be changed back to:
```suggestion
You already defined new models in your training. Did you check the access rights and define record rules on them? Are you sure your module is not an open door to SQL injections? Did you use `getattr` or `t-raw` in your code? Here are some things to which you should pay extra attention:
```